### PR TITLE
Make GB and UK regexes the same and add tests to ensure this

### PR DESCRIPTION
**Pull request by Onumis**
For https://github.com/dgilperez/validates_zipcode/issues/31

Ended up uniting GB and UK, so that they are now a great and truly united Britain/Kingdom!

Added a bunch of postcode that are valid (taken from postcodes.io), but not sure if enough.

### CODE REVIEW

File: lib/validates_zipcode/cldr_regex_collection.rb [hunk 1]
```diff
@@ -230,7 +230,7 @@ module CldrRegexpCollection
       TZ: /\A([a-zA-Z\d\s]){3,}\z/,
       TT: /\A\d{6}\z/,
       TW: /\A\d{3}\z/,
-      UK: /\A([A-PR-UWYZ0-9][A-HK-Y0-9][AEHMNPRTVXY0-9]?[ABEHMNPRVWXY0-9]? {1,2}[0-9][ABD-HJLN-UW-Z]{2}|GIR 0AA)\z/,
+      UK: /\AGIR[ ]?0AA|((AB|AL|B|BA|BB|BD|BH|BL|BN|BR|BS|BT|CA|CB|CF|CH|CM|CO|CR|CT|CV|CW|DA|DD|DE|DG|DH|DL|DN|DT|DY|E|EC|EH|EN|EX|FK|FY|G|GL|GY|GU|HA|HD|HG|HP|HR|HS|HU|HX|ID|IG|IM|IP|IV|JE|KA|KT|KW|KY|L|LA|LD|LE|LL|LN|LS|LU|M|ME|MK|ML|N|NE|NG|NN|NP|NR|NW|OL|OX|PA|PE|PH|PL|PO|PR|RG|RH|RM|S|SA|SE|SG|SK|SL|SM|SN|SO|SP|SR|SS|ST|SW|SY|TA|TD|TF|TN|TQ|TR|TS|TW|UB|W|WA|WC|WD|WF|WN|WR|WS|WV|YO|ZE)(\d[\dA-Z]?[ ]?\d[ABD-HJLN-UW-Z]{2}))|BFPO[ ]?\d{1,4}\z/, # Same as GB
```
Comment (dgilperez) [inline lib/validates_zipcode/cldr_regex_collection.rb]: This part scares me a little: are we sure we are maintaining backwards compatibility? At least in terms of whitespaces, dashes and such. I'm not familiar with UK codes, but not like to break current apps tendero y some formerly valid Codes as invalid.

What do you think?

File: spec/validates_zipcode_spec.rb [hunk 2]
```diff
@@ -263,6 +263,31 @@
     end
   end
 
+  ['UK', 'GB'].each do |cc|
+    context cc do
+      it 'validates with a valid zipcode' do
+        [
+          'ID1 1QD',
```
Comment (dgilperez) [inline spec/validates_zipcode_spec.rb]: Like I said above, what about omitting whitespaces or adding dashes? Are those valid?

**Comment by dgilperez** [comment]
So I looked deeper into this and I found an official regex provided by the GB goverment, which I ended up implementing for both `UK` and `GB`:

https://assets.publishing.service.gov.uk/government/uploads/system/uploads/attachment_data/file/488478/Bulk_Data_Transfer_-_additional_validation_valid_from_12_November_2015.pdf

Another funny thing is that `GB` is the official ISO country code, but there is an exception with `UK`, which is a reserved key for the same country. Hence, merging was the way to go. 

Thanks @Onumis 🍺 !

**Comment by tom-lord** [comment]
I just stumbled across this when updating my project's dependencies.

@dgilperez , you were right to be concerned about backwards compatibility. Whilst it *may* be valid to use that regex in favour of the old one, a major backwards incompatible change like this is not appropriate for a patch version release.

The new regex is in some ways much stricter, as it requires "properly formatted" whitespace. It is easy to find examples (with missing whitespace) of UK postcodes that were considered valid with the old format and invalid with this format. Furthermore, it's non-trivial to autocorrect postcodes to *include* whitespace (as opposed to the inverse operation, which is of course trivial).

I can't find a single consensus on what the "right regex" is. That government PDF is not an RFC. (It's not even entirely internally consistent.) For example, [wikipedia](https://en.wikipedia.org/wiki/Postcodes_in_the_United_Kingdom#Validation) suggests instead using:

```
^([A-Za-z][A-Ha-hJ-Yj-y]?[0-9][A-Za-z0-9]? ?[0-9][A-Za-z]{2}|[Gg][Ii][Rr] ?0[Aa]{2})$
```

(But it doesn't give a source for that regex.)

At this point, my recommendation - from a usability/backwards compatibility perspective - would be to just loosen the current regex to make the whitespace optional:

```
# CURRENT:
/\AGIR ?0AA|((([A-Z]\d{1,2})|(([A-Z][A-HJ-Y]\d{1,2})|(([A-Z]\d[A-Z])|([A-Z][A-HJ-Y]\d?[A-Z])))) \d[A-Z]{2})\z/i

# NEW:
/\AGIR ?0AA|((([A-Z]\d{1,2})|(([A-Z][A-HJ-Y]\d{1,2})|(([A-Z]\d[A-Z])|([A-Z][A-HJ-Y]\d?[A-Z])))) ?\d[A-Z]{2})\z/i
```

**Comment by dgilperez** [comment]
Thanks @tom-lord for raising this issue. I think you have a point about calling 0.0.19 a break of semver, if this is a breaking change (as it is) it should have been named 0.1.0. 

For the moment, I'm leaning towards another patch that fixes this by relaxing the regex as you suggest and mention that in the CHANGELOG. 

Let me see if I can release a quick patch - this may be hurting others as well.

**Pull request merged**